src/views/Pools/components/Stake.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { useState, useCallback } from 'react'
+import BigNumber from 'bignumber.js'
+import { useWeb3React } from '@web3-react/core'
+import { Button, Box, Flex, Text, useModal, IconButton, AddIcon, MinusIcon } from '@pancakeswap-libs/uikit'
+import UnlockButton from 'components/UnlockButton'
+import Balance from 'components/Balance'
+import { Pool } from 'state/types'
+import { useERC20 } from 'hooks/useContract'
+import useHasCakeBalance from 'hooks/useHasCakeBalance'
+import { useSousApprove } from 'hooks/useApprove'
+import { usePriceCakeBusd } from 'state/hooks'
+import useI18n from 'hooks/useI18n'
+import { getAddress } from 'utils/addressHelpers'
+import { getBalanceNumber } from 'utils/formatBalance'
+
+import TokenRequiredModal from './TokenRequiredModal'
+import StakeModal from './StakeModal'
+
+interface StakeProps {
+  pool: Pool
+  isOldSyrup: boolean
+  isBnbPool: boolean
+}
+
+const Stake: React.FC<StakeProps> = ({ pool, isOldSyrup, isBnbPool }) => {
+  const { userData, isFinished, stakingToken, sousId, stakingLimit, earningToken } = pool
+
+  const TranslateString = useI18n()
+  const cakePrice = usePriceCakeBusd()
+  const hasCake = useHasCakeBalance(new BigNumber(0))
+  const { account } = useWeb3React()
+  const stakingTokenContract = useERC20(stakingToken.address ? getAddress(stakingToken.address) : '')
+  const convertedLimit = new BigNumber(stakingLimit).multipliedBy(new BigNumber(10).pow(earningToken.decimals))
```

Comment:
we have some helpers in utils for tthis

Reply:
I took a look at this and it seems we dont have it?